salt/transport/zeromq.py

```diff
@@ -380,8 +380,9 @@ def __del__(self):
     def connect(self):
         if not self.auth.authenticated:
             yield self.auth.authenticate()
-        self.publish_port = self.auth.creds['publish_port']
-        self._socket.connect(self.master_pub)
+        if self.opts.get('detect_mode', False) is False:
+            self.publish_port = self.auth.creds['publish_port']
+            self._socket.connect(self.master_pub)
```

Comment:
Maybe I'm wrong, sorry if so. But I don't understand what this change for? What if we entered here in "detect" mode and `authenticate` call succeeded. In this case the channel is "authenticated" but it's socket is disconnected. Do you plan connect it later?